### PR TITLE
issue #11 and #15 - Delete-Button and Code selection

### LINKED ISSUE #11: #29 / Deletion of pastes manually
If we receive any information of unlawful posts we would need to be able to delete the pastes in question even before the automatic deletion after 12/24 hours.

### LINKED ISSUE #13: #28 / Setup RESTful routes for working with pastes
For this issue we're looking for an initial API design. The endpoints should be defined and return 501 Not Implemented status.

### PR DESCRIPTION
**Pull request by macskay**
Added Code Selection in paste
Added Delete Button

Resolves #11  and #15

### CODE REVIEW

File: pasteit/public/javascripts/globals.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+var mimes = {
+    "C++"  : "text/x-c++src",
+    "C"    : "text/x-csrc",
+    "C#"   : "text/x-csharp",
+    "Objective-C" : "text/x-objectivec",
+    "Java" : "text/x-java",
+};
+
+var getLangKey = function(value) {
```
Comment (ghost) [inline pasteit/public/javascripts/globals.js]: Change the value param to something more specific. Maybe languageName?

File: pasteit/public/javascripts/pasted.js [hunk 2]
```diff
@@ -5,6 +5,7 @@ $(document).ready(function() {
         mode = $('#mime-type').val();
     else
         mode = $("#selected-mode").val();
+    var humanreadable = getLangKey(mode);
```
Comment (ghost) [inline pasteit/public/javascripts/pasted.js]: Use consistent variable naming. This should be humanReadable, not humanreadable.

File: pasteit/public/javascripts/pasted.js [hunk 3]
```diff
@@ -16,4 +17,7 @@ $(document).ready(function() {
 
     editor.getDoc().setValue($("#pasted-content").val());
     editor.refresh();
+
+    $("#delete").prop("href", "/delete"+window.location.pathname);
```
Comment (ghost) [inline pasteit/public/javascripts/pasted.js]: Where's the linter man? spaces between + and the strings.

File: pasteit/views/pasted.ejs [hunk 4]
```diff
@@ -34,6 +35,12 @@
                 <div id="container" class="clearfix">
                     <div id="main-container">
                         <div id="paste-container">
+                            <div id="paste-header">
+                                <p>
+                                    <a id="delete" href="/delete/" style="float: right;">Delete</a>
```
Comment (ghost) [inline pasteit/views/pasted.ejs]: Inline styling needs to be moved to a CSS file.

**Review by ghost — CHANGES_REQUESTED** [review]
Just minor formatting stuff. Overall it looks good.

File: pasteit/public/javascripts/globals.js (hunk 1 above)
Comment (macskay) [reply pasteit/public/javascripts/globals.js]: done in 25e11d9

File: pasteit/public/javascripts/pasted.js (hunk 2 above)
Comment (macskay) [reply pasteit/public/javascripts/pasted.js]: done in 25e11d9

File: pasteit/public/javascripts/pasted.js (hunk 3 above)
Comment (macskay) [reply pasteit/public/javascripts/pasted.js]: done in 25e11d9

File: pasteit/views/pasted.ejs (hunk 4 above)
Comment (macskay) [reply pasteit/views/pasted.ejs]: done in 25e11d9

**Comment by macskay** [comment]
merged after applying changes

**Pull request merged**